src/Analyzers/Core/Analyzers/Helpers/DiagnosticHelper.cs

```diff
@@ -251,8 +253,28 @@ static string EncodeIndices(IEnumerable<int> indices, int additionalLocationsLen
                 descriptor.HelpLinkUri,
                 location,
                 additionalLocations,
-                descriptor.CustomTags,
+                GetEffectiveCustomTags(descriptor, notificationOption),
                 properties);
+
+            static IEnumerable<string> GetEffectiveCustomTags(DiagnosticDescriptor descriptor, NotificationOption2 notificationOption)
```

Comment:
This is the second core change in this PR that builds on top of #69843. If the diagnostic reported here had it's severity altered through `option_name = option_value:severity` editorconfig entry, then we add this new custom tag to the reported diagnostic which ensures that the compiler does not apply any further severity configurations from editorconfig/globalconfig files that can alter this severity.